src/Orleans.CodeGeneration.Build/build/Microsoft.Orleans.OrleansCodeGenerator.Build.targets

```diff
@@ -1,16 +1,39 @@
+    <GeneratorAssembly>$(OrleansCodeGeneratorAssembly)</GeneratorAssembly>
+  </PropertyGroup>
+    
+  <PropertyGroup Condition="'$(OrleansCodeGeneratorAssembly)' == ''">
+    <CoreAssembly>$(MSBuildThisFileDirectory)..\tasks\netcoreapp2.0\Orleans.CodeGeneration.Build.dll</CoreAssembly>
+    <FullAssembly>$(MSBuildThisFileDirectory)..\tasks\net461\Orleans.CodeGeneration.Build.exe</FullAssembly>
+    
+    <!-- Specify the assembly containing the MSBuild tasks. -->
+    <MSBuildIsCore Condition="'$(MSBuildRuntimeType)' == 'Core'">true</MSBuildIsCore>
+    <TaskAssembly Condition="'$(MSBuildIsCore)' == 'true'">$(CoreAssembly)</TaskAssembly>
+    <TaskAssembly Condition="'$(MSBuildIsCore)' != 'true'">$(FullAssembly)</TaskAssembly>
+
+    <!-- When the MSBuild host is full-framework, we defer to PATH for dotnet -->
+    <DotNetHost Condition="'$(MSBuildIsCore)' != 'true'">dotnet</DotNetHost>
+    
+    <!-- Specify the assembly containing the code generator. -->
+    <TargetIsCore Condition="$(TargetFramework.StartsWith('netcore')) or $(TargetFramework.StartsWith('netstandard'))">true</TargetIsCore>
+    <GeneratorAssembly Condition="'$(TargetIsCore)' == 'true'">$(CoreAssembly)</GeneratorAssembly>
+    <GeneratorAssembly Condition="'$(TargetIsCore)' != 'true'">$(FullAssembly)</GeneratorAssembly>
+  </PropertyGroup>
+  
   <PropertyGroup>
-    <TaskAssembly Condition="'$(OrleansCodeGeneratorAssembly)' != ''">$(OrleansCodeGeneratorAssembly)</TaskAssembly>
-    <IsCore Condition=" '$(MSBuildRuntimeType)' == 'Core' and '$(OrleansCodeGeneratorAssembly)' == ''">true</IsCore>
-    <TaskAssembly Condition=" '$(IsCore)' == 'true' and '$(TaskAssembly)' == ''">$(MSBuildThisFileDirectory)..\tasks\netcoreapp2.0\Orleans.CodeGeneration.Build.dll</TaskAssembly>
-    <TaskAssembly Condition=" '$(IsCore)' != 'true' and '$(TaskAssembly)' == ''">$(MSBuildThisFileDirectory)..\tasks\net461\Orleans.CodeGeneration.Build.exe</TaskAssembly>
     <CodeGenDirectory Condition="'$([System.IO.Path]::IsPathRooted($(IntermediateOutputPath)))' == 'true'">$(IntermediateOutputPath)</CodeGenDirectory>
     <CodeGenDirectory Condition="'$(CodeGenDirectory)' == ''">$(ProjectDir)$(IntermediateOutputPath)</CodeGenDirectory>
     <OutputFileName>$(CodeGenDirectory)$(TargetName).orleans.g.cs</OutputFileName>
     <CodeGeneratorEnabled Condition="'$(OrleansCodeGenPrecompile)'!='true' and '$(DesignTimeBuild)' != 'true'">true</CodeGeneratorEnabled>
   </PropertyGroup>
 
-  <UsingTask TaskName="Orleans.CodeGeneration.GetDotNetHost" AssemblyFile="$(TaskAssembly)" Condition="'$(CodeGeneratorEnabled)' == 'true' and '$(IsCore)' == 'true'" />
+  <UsingTask TaskName="Orleans.CodeGeneration.GetDotNetHost" AssemblyFile="$(TaskAssembly)" Condition="'$(CodeGeneratorEnabled)' == 'true' and '$(DotNetHost)' == '' and '$(MSBuildIsCore)' == 'true'" />
 
   <!-- This target is run just before Compile for an Orleans Grain Interface Project -->
   <Target Name="GenerateOrleansCode"
```

Comment:
Before the target is executed and we'd use the binaries defined in the variables set above for easier troubleshooting I'd add Exist checks for them. When we're depending on dotnet.exe being in the path, for the exist check it should be treated differently, so we can give better error message if dotnet.exe execution fails. Since we don't know the full path we can defer the error message after execution I think.

Reply:
With fresh head I looked into the possible solutions here and since the worst case scenario we can have here is that we're using ```dotnet``` from the path, and that will not be found. But in that case ```Exec``` task will fail anyway, in the other cases we're covered, since either we're resolving ```dotnet``` location or using the freshly built code generator assembly - when building Orleans itself - so I think we're good, **no update needed from my part LGTM**